src/dune_engine/action_builder.mli

```diff
@@ -242,6 +294,13 @@ module Make_exec (Build_deps : sig
 
   (** Check whether an alias is defined. *)
   val alias_exists : Alias.t -> bool Memo.Build.t
+
+  (** [execute_action facts act] execute an action. We expect [act] to observe
+      the facts [facts] about the world. *)
```

Comment:
```suggestion
      the [facts] about the world. *)
```

Reply:
Call sites of `execute_action` and `execute_action_stdout` might become a bit clearer if the `fact Dep.Map.t` argument is named, e.g. `execute_action ~observing_facts:foo`.

Reply:
Sounds good, I did that.